src/config.rs

```diff
@@ -552,17 +728,21 @@ impl Config {
 
     #[inline]
     pub fn apply_metrics(&self) {
-        let clusters = self.clusters.read();
-        crate::net::cluster::active_clusters().set(clusters.len() as i64);
-        crate::net::cluster::active_endpoints().set(clusters.num_of_endpoints() as i64);
+        // let Some(clusters) = self.clusters() else {
+        //     return;
+        // };
+        crate::metrics::apply_clusters(&self.clusters);
     }
 
     pub fn default_agent() -> Self {
         Self {
             clusters: Default::default(),
             filters: Default::default(),
-            id: default_proxy_id(),
-            version: Slot::with_default(),
+            dyn_cfg: DynamicConfig {
+                id: default_id(),
+                version: Version::default(),
+                typemap: default_typemap(),
+            },
```

Comment:
Is there a reason DynamicConfig doesn't impl default?

Reply:
Not really, will probably add it eventually.